### ISSUE TITLE
opencv from 4.5.3 to 4.5.4. cv::line not working with clang 12.0.1 but working with g++ 11.1.0

### ISSUE THREAD

**Issue by elegracer** [opencv/opencv#20854]
##### System information (version)
- OpenCV => 4.5.4
- Operating System / Platform => ArchLinux
- Compiler => clang 12.0.1

##### Detailed description

After upgrading opencv from 4.5.3 to 4.5.4, a simple `cv::line` program cannot work. Drawing a single line on a plain `cv::Mat` results in an unchanged image.

##### Steps to reproduce

```.cpp
#include <opencv2/opencv.hpp>

int main() {
    cv::Mat img(200, 200, CV_8UC3, CV_RGB(255, 255, 255));
    cv::arrowedLine(img, cv::Point(0, 0), cv::Point(100, 100), CV_RGB(255, 255, 0), 2, 8, 0);
    cv::imshow("img", img);
    cv::waitKey(0);
    return 0;
}
```

Compile it with clang 12.0.1 with the following command

```bash
clang++ -isystem /usr/include/opencv4 -O3 -DNDEBUG -fPIE -g -std=c++17 -o use_opencv use_opencv.cpp -lopencv_core -lopencv_features2d -lopencv_imgproc -lopencv_highgui
```

The shown image is plain white.

However, compile the same program with g++ 11.1.0, everything works as expected. A line is drawn on the white background.

```bash
g++ -isystem /usr/include/opencv4 -O3 -DNDEBUG -fPIE -g -std=c++17 -o use_opencv use_opencv.cpp -lopencv_core -lopencv_features2d -lopencv_imgproc -lopencv_highgui
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by pieniacy** [comment]
I can confirm this, pretty much the same environment (Arch, clang 12.0.1). The following **stopped working when upgrading 4.5.3 to 4.5.4**:
 - cv::circle
 - cv::putText
 - cv::arrowedLine
 - cv::drawContours

What still works:
 - cv::rectangle
 - cv::imshow, etc.

Functions do not show any error, they just silently do nothing. Above lists are not exhaustive, just what I noticed. Problem disappears when downgrading do 4.5.3 or when using gcc, none of which sounds like fun.

**Comment by cclauss** [comment]
Does this need to continue to be a pinned issue on https://github.com/opencv/opencv/issues ?

**Comment by mike-2020** [comment]
Has this issue been fixed? I can reproduce it in 4.5.5 and 4.6.

**Comment by lnexenl** [comment]
Similiar problem, OpenCV 4.5.4 cv::circle and cv::arrowedLine does not work with clang 17, but works with gcc 11
